### PR TITLE
W-14137848: Upgrade OTEL bom to the latest version

### PR DESCRIPTION
**Pull request by sofiamorseletto**

### CODE REVIEW

File: modules/metrics/mule-metrics-exporter-impl/pom.xml [hunk 1]
```diff
@@ -54,6 +54,20 @@
...
+                    <groupId>com.squareup.okio</groupId>
+                    <artifactId>okio</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+        <!-- Added back because this was brought transitively by okio -->
+        <dependency>
+            <groupId>com.squareup.okio</groupId>
+            <artifactId>okio-jvm</artifactId>
+            <version>3.2.0</version>
```
Comment (fsgonz) [inline modules/metrics/mule-metrics-exporter-impl/pom.xml:69]: Have you verified that this is the last version corresponding to the update in the otel pom?

File: modules/tracing/mule-tracer-exporter-impl/src/main/java/org/mule/runtime/tracer/exporter/impl/config/SpanExporterConfigurationUtils.java [hunk 2]
```diff
@@ -15,14 +15,14 @@
 import static java.lang.Long.parseLong;
 import static java.time.Duration.ofSeconds;
 
+import io.opentelemetry.sdk.common.export.RetryPolicy;
```
Comment (fsgonz) [inline modules/tracing/mule-tracer-exporter-impl/src/main/java/org/mule/runtime/tracer/exporter/impl/config/SpanExporterConfigurationUtils.java:18]: Order of import.

File: modules/tracing/mule-tracer-exporter-impl/pom.xml [hunk 3]
```diff
@@ -89,7 +85,11 @@
                 </exclusion>
             </exclusions>
         </dependency>
-        <!-- Added back because this was brought transitively by okio -->
+        <dependency>
```
Comment (fsgonz) [inline modules/tracing/mule-tracer-exporter-impl/pom.xml:88]: why is this needed?

**Review by fsgonz — CHANGES_REQUESTED** [review]

File: modules/tracing/mule-tracer-exporter-impl/pom.xml (hunk 3 above)
Comment (sofiamorseletto) [reply modules/tracing/mule-tracer-exporter-impl/pom.xml:88]: Which dependency do you mean?

Comment (fsgonz) [reply modules/tracing/mule-tracer-exporter-impl/pom.xml:88]: Now you don't have the exclusion opentelemetry-exporter-otlp-common but in opentelemetry-exporter-otlp. Is that correct?

File: modules/metrics/mule-metrics-exporter-impl/pom.xml (hunk 1 above)
Comment (sofiamorseletto) [reply modules/metrics/mule-metrics-exporter-impl/pom.xml:69]: Yes it's 3.2.0

File: modules/tracing/mule-tracer-exporter-impl/pom.xml (hunk 3 above)
Comment (sofiamorseletto) [reply modules/tracing/mule-tracer-exporter-impl/pom.xml:88]: Yes, on the new opentelemetry version, okio comes with `opentelemetry-exporter-otlp` instead of `opentelemetry-exporter-otlp-common` so I moved the exclusion from `opentelemetry-exporter-otlp-common` to `opentelemetry-exporter-otlp`

**Review by fsgonz — APPROVED** [review]

**Comment by sofiamorseletto** [comment]
--validate
